pubsublite/internal/wire/assigner.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package wire
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+
+	vkit "cloud.google.com/go/pubsublite/apiv1"
+	pb "google.golang.org/genproto/googleapis/cloud/pubsublite/v1"
+)
+
+// partitionSet is a set of partition numbers.
+type partitionSet map[int]struct{}
+
+func newPartitionSet(assignmentpb *pb.PartitionAssignment) partitionSet {
+	var void struct{}
+	partitions := make(map[int]struct{})
+	for _, p := range assignmentpb.GetPartitions() {
+		partitions[int(p)] = void
+	}
+	return partitionSet(partitions)
+}
+
+func (ps partitionSet) Ints() (partitions []int) {
+	for p := range ps {
+		partitions = append(partitions, p)
+	}
+	return
+}
+
+func (ps partitionSet) Contains(partition int) bool {
```

Comment:
nit. consider a typedef

Reply:
Ah yeah... I used raw types in the API due to other reviewers' comments and just kept it consistent in the wire package. Will think about defining strong types (in a separate PR, since it requires unrelated changes). Since this is called `partitionSet`, it might be ok.